.github/workflows/pull-request-management.yml

```
@@ -150,7 +150,7 @@ jobs:
         run: |
           cd tests/
           make unit-tests
-      - uses: actions/upload-artifact@v3
+      - uses: actions/upload-artifact@v4
```

In AVD we upload the artifact only for one python version (3.11) maybe we should do the same here?
It seems it is causing issue for this new version of upload-artifact
```suggestion
      - uses: actions/upload-artifact@v4
        if: |
          matrix.python == '3.11'
```

thats the 2nd breaking change described here between v3 and v4